Assets/MRTK/Core/Services/BaseDataProviderAccessCoreSystem.cs

```diff
@@ -203,12 +203,33 @@ public virtual IMixedRealityDataProvider GetDataProvider(string name)
 
             if (concreteType == null)
             {
-                Debug.LogError($"Unable to register {typeof(T).Name} data provider ({(!string.IsNullOrWhiteSpace(providerName) ? providerName : "unknown")}) because the value of concreteType is null.\n" +
-                    "This may be caused by code being stripped during linking. The link.xml file in the MixedRealityToolkit.Generated folder is used to control code preservation.\n" +
-                    "More information can be found at https://docs.unity3d.com/Manual/ManagedCodeStripping.html.");
+                if (!Application.isEditor)
+                {
+                    Debug.LogWarning($"Unable to register {typeof(T).Name} data provider ({(!string.IsNullOrWhiteSpace(providerName) ? providerName : "unknown")}) because the value of concreteType is null.\n" +
+                        "This may be caused by code being stripped during linking. The link.xml file in the MixedRealityToolkit.Generated folder is used to control code preservation.\n" +
+                        "More information can be found at https://docs.unity3d.com/Manual/ManagedCodeStripping.html.");
+                }
                 return false;
             }
 
+            SupportedUnityXRPipelines selectedPipeline =
+#if UNITY_2020_1_OR_NEWER
+                SupportedUnityXRPipelines.XRSDK;
+#elif UNITY_2019
+                !XRSettingsUtilities.IsLegacyXRActive ? SupportedUnityXRPipelines.XRSDK : SupportedUnityXRPipelines.LegacyXR;
```

Comment:
I guess this is a general comment rather than specific for this line, but I am wondering this changes in this PR can properly support scenarios where neither of the two pipelines are in use (e.g. maybe neither is used in the web verion of MRTK)?

Reply:
I think currently, the default behavior won't be any different than what happens today. I agree that we might want to consider that case though! Happy to track that with a follow-up issue